File: app/src/main/java/ru/andrey/cleandictionary/domain/model/Language.java
package ru.andrey.cleandictionary.domain.model;


public enum Language {
    RUSSIAN("ru"),
    ENGLISH("en"),
    FINNISH("fi");

    private final String languageCode;

    Language(String languageCode) {
        this.languageCode = languageCode;
    }

    public String getLanguageCode() {
        return languageCode;
    }

    public static Language byCode(String code) {
        for (Language language : Language.values()) {
            if (code.equalsIgnoreCase(language.getLanguageCode())) {
                return language;
            }
        }
        throw new IllegalArgumentException("Wrong code: " + code);
    }
}

File: domain/src/main/java/ru/andrey/domain/model/Language.kt
package ru.andrey.domain.model

enum class Language(val languageCode: String) {

    RUSSIAN("ru"),
    ENGLISH("en"),
    FINNISH("fi");

    companion object {

        fun byCode(code: String): Language {
            return values().find { code.equals(it.languageCode, ignoreCase = true) }
                    ?: throw IllegalArgumentException("Wrong code: $code")
        }
    }
}
